### ISSUE TITLE
Separate cache class...yes, again!

### ISSUE THREAD

**Issue by andrewtch** [twigphp/Twig#1415]
As for me, I see no need for memcache / redis / whatsoever cache in real life, yet I've stumbled upon a real-life situation where it is necessary:
- twig is used for a complex (8-10 levels of inheritance) system of landings
- landings are deployed automatically on read-only file system (no file cache)
- it is not allowed to have any directories writable by web server (since landings include partner scripts, this is a security measure)
- 'warming up' cache breaks continious deployment as it is now
- parsing templates takes considerable amount of cpu time

I can PR in a following way:
- add Twig_Cache_Interface
- add Twig_Cache_Filesystem mimicing current logic
-  alter Twig_Environment -> setCache call to automatically instantiate Twig_Cache_Filesystem($cache) if $cache is not instanceof Twig_Cache_Interface

that's all. Simply one cache, minor changes in loadTemplate, no cache - dependent implementations (We leave Twig_Cache_Whatsoever to user having such issues as this code should not be in a core cos mostly opcode cachers perform faster).

tl;dr: I want to refactor twig cache to support custom cache drivers, but not because of some heretic reasons like "memcache is faster than ZendOpcode", but for those users who have read-only filesystem but still want to save time on template parsing (like me).

PR can be done in some hours )
